tests/haskell-load-tests.el

```diff
@@ -0,0 +1,82 @@
+;;; haskell-load-tests.el
+
+;;; Code:
+
+(require 'ert)
+(require 'haskell-test-utils)
+
+(require 'haskell-load)
+
+(defun insert-errors ()
+  (insert "import           Control.Applicativ\nimport           Data.Mayb\nimport           Data.String")
+  (let ((a (make-overlay 18 36))
+        (b (make-overlay 54 63)))
+    (overlay-put a 'haskell-check t)
+    (overlay-put a 'haskell-msg-type 'error)
+    (overlay-put a 'haskell-msg "Could not find module ‘Control.Applicativ’\n    Perhaps you meant Control.Applicative (from base-4.8.1.0)\n    Use -v to see a list of the files searched for.")
+    (overlay-put b 'haskell-check t)
+    (overlay-put b 'haskell-msg-type 'error)
+    (overlay-put b 'haskell-msg "Could not find module ‘Data.Mayb’\n    Perhaps you meant\n      Data.Maybe (from base-4.8.1.0)\n      Data.Map (from containers-0.5.6.2@conta_LKCPrTJwOTOLk4OU37YmeN)\n    Use -v to see a list of the files searched for.")))
+
+(ert-deftest goto-first-error-before ()
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 1)
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
+
+(ert-deftest goto-first-error-after ()
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 92)
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
+
+(ert-deftest goto-first-error-between ()
+  (with-temp-switch-to-buffer
+   (insert-errors)
+   (goto-char 37)
+   (haskell-goto-first-error)
+   (should (equal (point) 18))))
```

Comment:
Please either use `re-search-forward` or use `looking-at-p` to check if we are in the correct spot in the source file.